### ISSUE TITLE
[BUG] after a multiclass classification setup, decision tree construction fails

### ISSUE THREAD

**Issue by zingochampis** [pycaret/pycaret#1520]
**Describe the bug**

After a multiclass classification setup, a `create_model('dt')` fails. Something there defaults to an inapplicable parameter.
Here is the error message:

**File "C:\ProgramData\Anaconda3\lib\site-packages\sklearn\metrics\_classification.py", line 1263, in _check_set_wise_labels
    raise ValueError("Target is %s but average='binary'. Please "
ValueError: Target is multiclass but average='binary'. Please choose another average setting, one of [None, 'micro', 'macro', 'weighted'].**

**To Reproduce**

```python
from pycaret.classification import *
s = setup(data=df1,
          normalize = True, 
#                     pca=True,
                  transformation = True,                
          # remove_multicollinearity = True,
         #  multicollinearity_threshold = 0.70,
           feature_interaction=True,
#           ignore_low_variance = True,
 numeric_features =  ['Risk observation', 
                      'NA',
                      'Injury', 
                      'Near misses',  
                     # 'ro_1',
                      'na_1', 
                      'i_1',
                      'nm_1', 
                      #'ro_2',
                      'na_2',
                      'i_2',
                      'nm_2', 
                      'Client', 'Internal', 'BDI', 'cl_1', 'in_1', 'BDI_1', 'cl_2', 'in_2','BDI_2'
                      #,  
                      #'avg%', 'goodmarks%',  
                      #'Scheduled hours', 'Sick leave hours TOTAL', 'Sick leave %', '1-3 days', '1-3 days%',
                      #'short_under_16days', 'short_under_16days%', 'long_over_16days', 'long%'
                     ], 
target='class',
group_features = [['Injury','i_1','i_2',
                   'Risk observation',# 'ro_1','ro_2',
                   'Near misses','nm_1','nm_2'],
                 ['Internal','in_1','in_2','BDI','BDI_1','BDI_2','Client','cl_1','cl_2']],
group_names=['NonConf','Improv'],
#fix_imbalance=True,fix_imbalance_method= oversampler,
          data_split_stratify=True, session_id =15, silent=True)

dt=create_model('dt')
```

**Expected behavior**

model is created and some metrics are shown. it is possible to plot a multiclass confusion matrix.

**Additional context**

**Versions**
<details>

</details>

**Comment by moezali1** [comment]
@zingochampis Thanks for reporting. I am unable to reproduce the error on `iris` dataset.

```
from pycaret.datasets import get_data
data = get_data('iris')

from pycaret.classification import *
s = setup(data, target = 'species')

dt = create_model('dt')
```

![image](https://user-images.githubusercontent.com/54699234/129983349-411ef3ec-d1a4-40ea-9336-1ed6a4bed7d7.png)

Do you think you can share the dataset? Maybe I can help.

**Comment by zingochampis** [comment]
[complaint2.csv](https://github.com/pycaret/pycaret/files/7012918/complaint2.csv)
Attached: dataframe that is used here.
```
df1= pd.read_csv('complaint2.csv',sep='|')
```

**Comment by moezali1** [comment]
Thanks @zingochampis. This is happening because one of your class only contains 5 example and the number of folds for cross validation is by default set to 10. 

![image](https://user-images.githubusercontent.com/54699234/130103702-87086f1f-757f-459a-bae3-f445b2fc5c5a.png)

To fix the problem on this dataset I suggest you either get rid of class 2 which only contains 5 examples or your number of folds must be less than 5. This is how I fixed it:

```
data = pd.read_csv('complaint2.csv', sep='|')
from pycaret.classification import *
s = setup(data, target = 'class', fold = 3)
dt = create_model('dt')
```

You can alternatively not pass the `fold` parameter in the `setup` and instead just pass them in individual functions like `create_model('dt', fold = 3). This will have the same effect.

![image](https://user-images.githubusercontent.com/54699234/130104154-2bd41bc0-081f-45ca-9f6b-d465f86a839e.png)

Hope this helps.

@Yard1 While the problem is fixed, I am not sure the we are getting this particular ValueError in first place.

ValueError: Target is multiclass but average='binary'. Please choose another average setting, one of [None, 'micro', 'macro', 'weighted'].

This seems like an edge case. I also remember we once had an Exception clause that will catch if any class has less instances the number of fold at the `setup` execution. I think we can catch this. Any thoughts?

**Comment by moezali1** [comment]
@Yard1 If this is a quick fix then we can take this in `2.3.4`
